src/utils/getDiscordIds.ts

```diff
@@ -0,0 +1,70 @@
+import { RDS_BASE_API_URL } from "../constants/urls";
+import { UserBackend } from "../typeDefinitions/userBackend.types";
+import * as response from "../constants/responses";
+
+export const getDiscordIds = async (
+  userIds: string[] | string
+): Promise<string[] | string> => {
+  try {
+    const url = `${RDS_BASE_API_URL}/users/userId`;
+
+    const numberOfBatches = Math.ceil(userIds.length / 6);
+
+    const fetchCollection = [] as Array<Promise<Response>[]>;
+
+    //for storing the data returned
+    const fetchResponseCollection = [] as Array<UserBackend[]>;
+
+    let indexCollection = 0;
+    let numberOfFetchCalls = 0;
+
+    const discordIds = [] as Array<string>;
+
+    //adding all batch
+    for (let i = 0; i < numberOfBatches; i++) {
+      const batch: Promise<Response>[] = [];
+      fetchCollection.push(batch);
```

Comment:
Can you please explain what's being done here? This is not very evident. Kindly change variable names

Reply:
We are creating a batch, which consists 6 fetch calls, and fetchCollection is storing those batches

Reply:
I have already calculated the number of batches needed, so i am initialising these batches and storing them in fetchCollection